cmd/kubeless/function/function.go

```diff
@@ -133,7 +134,7 @@ func getContentType(filename string, fbytes []byte) string {
 	return contentType
 }
 
-func getFunctionDescription(cli kubernetes.Interface, funcName, ns, handler, file, deps, runtime, runtimeImage, mem, cpu, timeout string, envs, labels []string, secrets []string, defaultFunction kubelessApi.Function) (*kubelessApi.Function, error) {
+func getFunctionDescription(cli kubernetes.Interface, funcName, ns, handler, file, deps, runtime, runtimeImage, mem, cpu, timeout string, port int32, headless bool, envs, labels []string, secrets []string, defaultFunction kubelessApi.Function) (*kubelessApi.Function, error) {
```

Comment:
What's the usecase for varying port/headless here?

I think the Service we're creating here is an internal trigger->function implementation detail, so altering the port doesn't make much sense - and if making it headless is better/required in some way, then surely we should do that unconditionally for all Functions...

In particular, if we allow changing this here, then I think *every* trigger needs to be modified to deal with surprise values of ports, and to ensure that load is still balanced across replicas appropriately for headless services.  I raise this is a reason that we should *not* support this variation here.

Reply:
> It is not so "internal".

Rather than exposing this internal detail, and thus requiring it to be more flexible than is needed purely for kubeless' own use, and thus pushing complexity onto all the other parts of kubeless - I propose we keep the internal Service as internal and _don't_ support this flexibility here.

If users *want* a headless Service, or a different port, or some other functionality (type=LoadBalancer?), then I propose we instead ask those users to create a *new* Service with whatever special bits they want, and just use the same label selector + targetPort=8080.   We could even create a ServiceTrigger that did that for them...

Reply:
If no one wants (lets check in the slack and community call) may be we should just remove it. Its  been pain to keep supporting it.